openassessment/xblock/openassessmentblock.py

```diff
@@ -364,12 +364,87 @@ def student_view(self, context=None):
+                      "url_grade_available_responses": "/grade_available_responses_view",
+                      "staff_assessment": false,
+                      "parent_id": "vertical_block_id",
+                      "url_base": "/student_view",
+                      "id": "openassessment_block_id"
+                     }, ...]
+                "ora_item_view_enabled" - enabled LMS API endpoint to serve XBlock view or not
+
+        Returns:
+            (Fragment): The HTML Fragment for this XBlock.
+        """
+        ora_items = context.get('ora_items', []) if context else []
+        ora_item_view_enabled = context.get('ora_item_view_enabled', False) if context else False
+        context_dict = {
+            "ora_items": json.dumps(ora_items),
+            "ora_item_view_enabled": 1 if ora_item_view_enabled else 0
+        }
+
+        template = get_template('openassessmentblock/instructor_dashboard/oa_listing.html')
+        context = Context(context_dict)
+        fragment = Fragment(template.render(context))
+
+        return self._update_and_return_fragment(fragment, 'CourseOpenResponsesListingBlock')
+
+    def grade_available_responses_view(self, context=None):
+        """Grade Available Responses view.
+
+        Auxiliary view which displays the staff grading area
+        (used in the Open Response Assessment tab in the Instructor Dashboard of LMS)
+
+        Args:
+            context: Not used for this view.
+
+        Returns:
+            (Fragment): The HTML Fragment for this XBlock.
+        """
+        student_item = self.get_student_item_dict()
+        staff_assessment_required = "staff-assessment" in self.assessment_steps
+
+        context_dict = {
+            "title": self.title,
+            'staff_assessment_required': staff_assessment_required
+        }
+
+        if staff_assessment_required:
+            context_dict.update(
+                self.get_staff_assessment_statistics_context(student_item["course_id"], student_item["item_id"])
+            )
+
+        template = get_template('openassessmentblock/instructor_dashboard/oa_grade_available_responses.html')
+        context = Context(context_dict)
+        fragment = Fragment(template.render(context))
+
+        return self._update_and_return_fragment(fragment, 'StaffAssessmentBlock')
+
+    def _update_and_return_fragment(self, fragment, initialize_js_func):
+        """
+        Auxiliary function to return updated fragment
+
+        """
         i18n_service = self.runtime.service(self, 'i18n')
         if hasattr(i18n_service, 'get_language_bidi') and i18n_service.get_language_bidi():
             css_url = "static/css/openassessment-rtl.css"
         else:
             css_url = "static/css/openassessment-ltr.css"
 
+        # backgrid.js v0.3.7
+        fragment.add_css_url(self.runtime.local_resource_url(self, "static/css/lib/backgrid.min.css"))
+        self.add_javascript_files(fragment, "static/js/lib/backgrid.min.js")
```

Comment:
@bjacobel I agree that it is necessary to add non-minified version for development (and I'll add it) but I disagree with you about using `Backgrid.js` version from NPM. Theoretically - yes, you're right. But in practice it doesn't work. This approach is working **only for tests** that are executed from the ORA projects. The links that you posted designed for LMS but here is XBlock.

When LMS install XBlock dependencies it uses only existed files (that are described in the [MANIFEST.in](https://github.com/edx/edx-ora2/blob/master/MANIFEST.in)). So you can't run `npm install` during install all these dependencies from pip. Of course I may add `Backgrid.js` to the minified version `openassessment-lms.min.js` by modifying this string:
https://github.com/edx/edx-ora2/blob/master/Makefile#L33
But it will work only for `settings.DEBUG == False`. In case of `settings.DEBUG == True` LMS requires non-minified versions: https://github.com/edx/edx-ora2/blob/master/openassessment/xblock/openassessmentblock.py#L373

Also I don't quite understand why this problem is so critical because as I may see there are a lot of dependencies that are used from the local folder (not from NPM). Examples:
https://github.com/edx/edx-ora2/tree/master/openassessment/xblock/static/js/lib
https://github.com/edx/edx-platform/tree/master/common/static/js/vendor

/cc @efischer19